third-party/realdds/include/realdds/topics/image/image-msg.h

```diff
@@ -34,14 +43,30 @@ class image
 public:
     using type = raw::device::imagePubSubType;
 
+    image() = default;
     image( const raw::device::image & );
+    image( const image & );
+    image( image && );
+
+    image & operator=( image && rhs );
+    image & operator=( const raw::device::image & );
 
     bool is_valid() const { return width != 0 && height != 0; }
     void invalidate() { width = 0; }
 
     static std::shared_ptr< dds_topic > create_topic( std::shared_ptr< dds_participant > const & participant,
                                                       char const * topic_name );
 
+    // This helper method will take the next sample from a reader. 
+    // 
+    // Returns true if successful. Make sure you still check is_valid() in case the sample info isn't!
+    // Returns false if no more data is available.
+    // Will throw if an unexpected error occurs.
+    //
+    static bool take_next( dds_topic_reader &,
```

Comment:
Please add to the comment that this helper function performs a copy and TODO:: add a `take_next_sample` method to avoid this copy in the future

Reply:
I added the comment for now, but I think that we don't need two API functions and will always use the non-copy one